File: android/app/src/krk/java/ru/ubsmobile/BankTheme/BankTheme.java
package ru.ubsmobile.BankTheme;

import com.facebook.react.bridge.ReactApplicationContext;

public class BankTheme extends BankThemeBase {

    public BankTheme(ReactApplicationContext context) {
        super(context);
    }

    public static String BackgroundColor = "#24207D";
    @Override
    protected String Color1() {
        return "#24207D";
    }

    public void requestAutostartPermission() {
        super.requestAutostartPermission();
    }

    @Override
    protected boolean AllowShowCvvCode() { return false; }

    @Override
    protected String NavigationBackgroundColor() {
        return "#24207D";
    }

    @Override
    protected MapLocation DefaultMapLocation() {
        return new MapLocation("55.824430", "37.388902");
    }

    @Override
    protected boolean ShowBonusesPage() {
        return false;
    }

    @Override
    protected boolean ShowNotificationsSetting() {
        return true;
    }

    @Override
    protected boolean AllowAddCardsToWallet() {
        return false;
    }

    @Override
    protected boolean ShowPersonalOffersOnMyBankPage() { return true; }

    @Override
    protected CodeSettingPageTheme CodeSettingPageTheme() {
        return null;
    }

    @Override
    protected String BankPhoneNumber() { return "+74952281244"; }

    @Override
    protected boolean PushNotificationsUsed() { return false; }

    @Override
    protected boolean ShowQrCodeIcon() { return false; }

    @Override
    protected boolean BankMessagesUsed() { return true; }

}

File: ios/rub_app/BankTheme.swift
public class BankTheme: NSObject, BankThemeBase {
  public var udSuiteName: String = "group.rub.notificationsService"
  public var color1: String = "#0970cd";
  public var serverUrl: String = "https://212.48.39.11:11443";
  public var showNotificationsSetting = false
  public var showQrCodeIcon = false
  public var showBonusesPage: Bool = false
  public var statusBarTheme: String = "light"
  public var navigationBackgroundColor: String = "#0970cd"
  public var allowAddCardsToWallet = false
  public var allowShowCvvCode = false
  public var showPersonalOffersOnMyBankPage = false
  public var bankPhoneNumber: String = "+74956483687"
  public var pushNotificationsUsed: Bool = false;
  public var bankMessagesUsed: Bool = true;

  public var defaultMapLocation: MapLocation = MapLocation(latitude: "55.761099", longitude: "37.654990")

  public var codeSettingPageTheme: CodeSettingPageTheme?

  override init() {
    self.codeSettingPageTheme = CodeSettingPageTheme(backgroundColor: self.color1,
                                       buttonBackgroundColor: "white",
                                       buttonBorderColor: self.color1,
                                       buttonTextColor: self.color1,
                                       textColor: "white",
                                       indicatorBorderColor: "white",
                                       indicatorEmptyBackgroundColor: "transparent",
                                       indicatorFullBackgroundColor: "white"
    )
  }
}
